src/Illuminate/Database/Connectors/MySqlConnector.php

```diff
@@ -104,24 +104,51 @@ protected function getHostDsn(array $config)
     }
 
     /**
+     * Get the modes for the connection.
+     *
+     * @param  array  $config
+     * @return array
+     */
+    protected function getModes(array $config)
+    {
+        if (isset($config['modes'])) {
+            return (array) $config['modes'];
```

Comment:
What's the purpose of the array cast? If it's to allow someone to do `'modes' => 'SOME_MODE'` when they only want to enable one mode vs. `'modes' => ['SOME_MODE']` I think it's probably simpler to just expect everyone to use an array.